Sources/BraintreeUIComponents/VenmoButton.swift

```diff
@@ -58,27 +64,51 @@ public struct VenmoButton: View {
         self.completion = completion
         self.apiClient = BTAPIClient(authorization: authorization)
     }
+
     public var body: some View {
         PaymentButtonView(
             color: color ?? .blue,
             width: width,
             logoHeight: 14,
             accessibilityLabel: "Pay with Venmo",
-            accessibilityHint: "Complete payment using Venmo"
+            accessibilityHint: "Complete payment using Venmo",
+            isDisabled: isLoading,
+            spinnerImageName: color?.spinnerColor,
+            isLoading: isLoading,
+            spinnerRotation: spinnerRotation,
         ) {
             apiClient?.sendAnalyticsEvent(UIComponentsAnalytics.venmoButtonSelected)
+            isLoading = true
+            spinnerRotation = 0
             invokeVenmoFlow()
         }
         .onAppear {
             apiClient?.sendAnalyticsEvent(UIComponentsAnalytics.venmoButtonPresented)
+            isLoading = false
+        }
+        // Spinner animation
+        .onChange(of: isLoading) { _, loading in
```

Comment:
Do we need to add the other states here since Venmo also supports universal links?